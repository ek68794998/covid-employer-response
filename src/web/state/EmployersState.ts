import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}