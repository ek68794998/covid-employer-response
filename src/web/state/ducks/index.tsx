import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});