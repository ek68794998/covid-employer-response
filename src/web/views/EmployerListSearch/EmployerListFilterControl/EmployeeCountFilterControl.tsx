import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";

import { LocalizedStrings } from "../../../../common/LocalizedStrings";

import { getStrings } from "../../../state/ducks/localization/selectors";

import EmployerListSearchFiltersPopup, { SelectorProps } from "../../EmployerListSearchFiltersPopup/EmployerListSearchFiltersPopup";

import { EmployerListSearchFilterProps } from "../EmployerListSearchFilterProps";

import "./EmployerListFilterControl.scss";

const EmployeeCountFilterControl: React.FC<EmployerListSearchFilterProps> =
	(props: EmployerListSearchFilterProps): React.ReactElement => {
		const strings: LocalizedStrings = useSelector(getStrings);

		const { initialFilter, onUpdateFilterValue } = props;

		const [ showSmall, setShowSmall ] = useState(initialFilter.small);
		const [ showMedium, setShowMedium ] = useState(initialFilter.medium);
		const [ showLarge, setShowLarge ] = useState(initialFilter.large);
		const [ isPopupOpen, setIsPopupOpen ] = useState(false);
		const [ displayText, setDisplayText ] = useState(strings.filters.employeesDefault);

		const clear = (e: React.MouseEvent<HTMLElement>): void => {
			setShowSmall(true);
			setShowMedium(true);
			setShowLarge(true);

			e.stopPropagation();
		};

		useEffect(
			(): void => {
				let newDisplayText: string;

				if (showSmall && showMedium && showLarge) {
					newDisplayText = strings.filters.employeesDefault;
				} else {
					const displayValues: string[] = [];

					if (showSmall) {
						displayValues.push(strings.filters.employeesSmall);
					}

					if (showMedium) {
						displayValues.push(strings.filters.employeesMedium);
					}

					if (showLarge) {
						displayValues.push(strings.filters.employeesLarge);
					}

					newDisplayText = displayValues.join(", ");
				}

				onUpdateFilterValue({ small: showSmall, medium: showMedium, large: showLarge });
				setDisplayText(newDisplayText);
			},
			[ showSmall, showMedium, showLarge ]);

		const children: SelectorProps[] = [
			{
				initialValue: showSmall,
				label: strings.filters.employeesSmall,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => setShowSmall(e.target.checked),
			},
			{
				initialValue: showMedium,
				label: strings.filters.employeesMedium,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => setShowMedium(e.target.checked),
			},
			{
				initialValue: showLarge,
				label: strings.filters.employeesLarge,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => setShowLarge(e.target.checked),
			},
		];

		const popup: JSX.Element | null =
			isPopupOpen
				? (
					<EmployerListSearchFiltersPopup
						childProps={children}
						multiselect={true}
						onClose={(): void => setIsPopupOpen(false)}
					/>
				)
				: null;

		const isActive: boolean = !(showSmall && showMedium && showLarge);

		const buttonClasses: string[] = [ "EmployerListFilterControl__Button" ];

		if (isActive) {
			buttonClasses.push("EmployerListFilterControl__Button--Active");
		}

		return (
			<div className="EmployerListFilterControl__Container">
				<button className={buttonClasses.join(" ")} onClick={(): void => setIsPopupOpen(!isPopupOpen)}>
					{isActive && <i className="material-icons" onClick={clear}>clear</i>}
					<div>{displayText}</div>
				</button>
				{popup}
			</div>
		);
	};

export default EmployeeCountFilterControl;
