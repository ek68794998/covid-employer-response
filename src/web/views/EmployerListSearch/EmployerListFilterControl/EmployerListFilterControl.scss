@import "../../Colors.scss";
@import "../../Dimensions.scss";

$filter-button-height: 32px;

.EmployerListFilterControl__Container {
	position: relative;
}

.EmployerListFilterControl__Button {
	background: $background;
	border: 1px solid $outline-color;
	border-radius: ($filter-button-height / 2);
	color: $link-hover;
	cursor: pointer;
	height: $filter-button-height;
	outline: none;
	padding: 0 ($filter-button-height / 3);
	white-space: nowrap;
	width: 120px;

	&.EmployerListFilterControl__Button--Active {
		background: $link-background;
		border-color: transparent;
		padding-left: 4px;

		i {
			border-radius: 50%;
			float: left;
			font-size: 18px;
			margin: 2px 1px;
			padding: 2px;

			&:hover {
				cursor: pointer;
				background: $link-normal;
			}
		}
	}

	div {
		overflow: hidden;
		padding: 1px;
		text-overflow: ellipsis;
	}
}
