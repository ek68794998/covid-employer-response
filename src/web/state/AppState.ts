import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}