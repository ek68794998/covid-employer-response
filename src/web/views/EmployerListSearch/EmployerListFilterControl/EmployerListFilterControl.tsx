import React, { useState } from "react";
import { RouteProps } from "react-router-dom";

import EmployerListSearchFiltersPopup, { SelectorProps } from "../../EmployerListSearchFiltersPopup/EmployerListSearchFiltersPopup";

import "./EmployerListFilterControl.scss";

export interface ChildSelectorProps {
	initialValue: boolean;

	label: string;

	setValue: (value: boolean) => void;
}

interface Props extends RouteProps {
	getDisplayText: () => string;

	isActive: boolean;

	onClear: () => void;

	popupTitle: string;

	selectorPropsList: ChildSelectorProps[];
}

const EmployerListFilterControl: React.FC<Props> =
	(props: Props): React.ReactElement => {
		const { getDisplayText, isActive, onClear, popupTitle, selectorPropsList } = props;

		const [ isPopupOpen, setIsPopupOpen ] = useState(false);
		const [ displayText, setDisplayText ] = useState(getDisplayText());

		const clear = (e: React.MouseEvent<HTMLElement>): void => {
			onClear();
			setDisplayText(getDisplayText());
			e.stopPropagation();
		};

		const children: SelectorProps[] =
			selectorPropsList.map((value: ChildSelectorProps): SelectorProps => ({
				initialValue: value.initialValue,
				label: value.label,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => {
					value.setValue(e.target.checked);
					setDisplayText(getDisplayText());
				},
			}));

		const popup: JSX.Element | null =
			isPopupOpen
				? (
					<EmployerListSearchFiltersPopup
						childProps={children}
						multiselect={true}
						onClose={(): void => setIsPopupOpen(false)}
						title={popupTitle}
					/>
				)
				: null;

		const buttonClasses: string[] = [ "EmployerListFilterControl__Button" ];

		if (isActive) {
			buttonClasses.push("EmployerListFilterControl__Button--Active");
		}

		return (
			<div className="EmployerListFilterControl__Container">
				<button className={buttonClasses.join(" ")} onClick={(): void => setIsPopupOpen(!isPopupOpen)}>
					{isActive && <i className="material-icons" onClick={clear}>clear</i>}
					<div>{displayText}</div>
				</button>
				{popup}
			</div>
		);
	};

export default EmployerListFilterControl;
