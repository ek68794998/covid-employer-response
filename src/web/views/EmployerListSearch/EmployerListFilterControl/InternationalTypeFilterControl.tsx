import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";

import { LocalizedStrings } from "../../../../common/LocalizedStrings";

import { getStrings } from "../../../state/ducks/localization/selectors";

import EmployerListSearchFiltersPopup, { SelectorProps } from "../../EmployerListSearchFiltersPopup/EmployerListSearchFiltersPopup";

import { EmployerListSearchFilterProps } from "../EmployerListSearchFilterProps";

import "./EmployerListFilterControl.scss";

const InternationalTypeFilterControl: React.FC<EmployerListSearchFilterProps> =
	(props: EmployerListSearchFilterProps): React.ReactElement => {
		const strings: LocalizedStrings = useSelector(getStrings);

		const [ showNational, setShowNational ] = useState(false);
		const [ showInternational, setShowInternational ] = useState(false);
		const [ isPopupOpen, setIsPopupOpen ] = useState(false);
		const [ displayText, setDisplayText ] = useState(strings.filters.locationDefault);

		const { onUpdateFilterValue } = props;

		const clear = (e: React.MouseEvent<HTMLElement>): void => {
			setShowNational(false);
			setShowInternational(false);

			e.stopPropagation();
		};

		useEffect(
			(): void => {
				let newDisplayText: string;

				if (showNational && !showInternational) {
					newDisplayText = strings.filters.locationNational;
				} else if (showInternational && !showNational) {
					newDisplayText = strings.filters.locationInternational;
				} else {
					newDisplayText = strings.filters.locationDefault;
				}

				onUpdateFilterValue({ international: showInternational, national: showNational });
				setDisplayText(newDisplayText);
			},
			[ showNational, showInternational ]);

		const children: SelectorProps[] = [
			{
				label: strings.filters.locationNational,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => setShowNational(e.target.checked),
			},
			{
				label: strings.filters.locationInternational,
				onChange: (e: React.ChangeEvent<HTMLInputElement>): void => setShowInternational(e.target.checked),
			},
		];

		const popup: JSX.Element | null =
			isPopupOpen
				? (
					<EmployerListSearchFiltersPopup
						childProps={children}
						multiselect={true}
						onClose={(): void => setIsPopupOpen(false)}
					/>
				)
				: null;

		const isActive: boolean = showNational !== showInternational;

		const buttonClasses: string[] = [ "EmployerListFilterControl__Button" ];

		if (isActive) {
			buttonClasses.push("EmployerListFilterControl__Button--Active");
		}

		return (
			<div className="EmployerListFilterControl__Container">
				<button className={buttonClasses.join(" ")} onClick={(): void => setIsPopupOpen(!isPopupOpen)}>
					{isActive && <i className="material-icons" onClick={clear}>clear</i>}
					<div>{displayText}</div>
				</button>
				{popup}
			</div>
		);
	};

export default InternationalTypeFilterControl;
