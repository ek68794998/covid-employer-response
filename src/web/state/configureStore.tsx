import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;