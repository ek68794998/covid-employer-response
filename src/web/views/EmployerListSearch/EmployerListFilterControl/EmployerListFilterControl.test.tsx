import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { AppState } from "../../../state/AppState";
import configureStore from "../../../state/configureStore";

import EmployerListFilterControl from "./EmployerListFilterControl";

describe("<EmployerListFilterControl />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListFilterControl
							getDisplayText={(): string => "δïƨƥℓá¥Tèжƭ"}
							isActive={true}
							onClear={(): void => { /* Do nothing. */ }}
							popupTitle="ƭïƭℓèTèжƭ"
							selectorPropsList={[]}
						/>
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
