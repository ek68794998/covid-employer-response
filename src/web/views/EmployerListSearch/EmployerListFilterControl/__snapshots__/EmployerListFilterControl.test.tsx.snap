// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListFilterControl /> renders without exploding 1`] = `
<div
  className="EmployerListFilterControl__Container"
>
  <button
    className="EmployerListFilterControl__Button EmployerListFilterControl__Button--Active"
    onClick={[Function]}
  >
    <i
      className="material-icons"
      onClick={[Function]}
    >
      clear
    </i>
    <div>
      δïƨƥℓá¥Tèжƭ
    </div>
  </button>
</div>
`;
